userspace/libsinsp/events/sinsp_events_ppm_sc.cpp

```diff
@@ -243,7 +244,18 @@ libsinsp::events::set<ppm_sc_code> libsinsp::events::names_to_sc_set(const std::
 			ppm_sc_set.insert((ppm_sc_code)ppm_sc);
 		}
 	}
-	return ppm_sc_set;
+	/* Extra back and forth mapping to resolve overloaded event <-> sc names, e.g. accept -> accept, accept4
+	 * Plus account for variants that share event codes, e.g. eventfd, eventfd2 share PPME_SYSCALL_EVENTFD_E, PPME_SYSCALL_EVENTFD_X
+	 * Plus handle special snowflakes, e.g. "umount" event string maps to PPME_SYSCALL_UMOUNT_E, PPME_SYSCALL_UMOUNT_X, but
+	 * in actuality applies for "umount2" syscall as "umount" syscall is a generic event -> end result is activating both umount, umount2
+	 *
+	 * Since names_to_event_set would resolve generic sc events, we only apply these extra lookups for non generic sc event codes
+	*/
+	auto all_non_generic_sc_event_set = libsinsp::events::all_event_set().filter([&](ppm_event_code e) { return libsinsp::events::is_syscall_event(e); })\
+	.diff(libsinsp::events::set<ppm_event_code>{PPME_GENERIC_E, PPME_GENERIC_X});
+	auto tmp_event_set = all_non_generic_sc_event_set.intersect(libsinsp::events::names_to_event_set(syscalls));
+	auto tmp_sc_set = libsinsp::events::event_set_to_sc_set(tmp_event_set);
+	return ppm_sc_set.merge(tmp_sc_set);
```

Comment:
I'm ok with this because I see the reason behind it, but I would prefer fixing the tables (e.g. the `umount` corner case) and iterate back here in the future to make this function simpler and less defensive. Just leaving a TODO for us.

Reply:
Hi @jasondellaluce thank you for your additional input. 

The primary reasoning behind these changes that follow the principle of **right-sized incremental progress** are as follows:

- We benefit from a complete and technically correct solution without any additional major refactor. A major refactor tends to always carry unknowns.
- We now have time to reflect on what will be best for the long-term project roadmap as per our project governance without blocking crucial testing of the new Falco feature https://github.com/falcosecurity/falco/pull/2432. There is always a possibility that while a new table appears better at the moment to cover these corner cases, we however could then discover the reasoning why it wasn't done that way in the first place. Objectively these two tables have been around almost from the beginning of the project, so would vote for first gathering more context. In summary, in order to attempt making the right decisions allowing for more time to think seems a fair choice.

See also https://github.com/falcosecurity/libs/pull/889#issuecomment-1448439205

Reply:
not sure this is what we want, I need to double-check, my brain with all this PPM stuff is like -> :exploding_head:

Reply:
Falco rule calls "accept" -> should apply filter to both accept, accept4
Falco rule calls "eventfd" -> should apply filter to both eventfd, eventfd2

This is the behavior this change ensures without requiring a table refactor.

Reply:
ok, I got the point thank you for that!